### ISSUE TITLE
TTD: breakpoints not working without a debugger statement in the code

### ISSUE THREAD

**Issue by mmarchini** [nodejs/node-chakracore#516]
* **Version**: chakracore-nightly (from nvs)
* **Platform**: Mac OS X (10.13.4)
* **Subsystem**: TTD

Is TTD working on Mac? I tried using chakracore-nighly from nvs + VS Code 1.22.2 but breakpoints and "go back in time" are not working when replaying the trace (the replay just stops when I try to use either feature). I'm following instruction from [here](https://github.com/nodejs/node-chakracore/blob/master/TTD-README.md).

![kapture 2018-04-18 at 18 37 38](https://user-images.githubusercontent.com/4048656/38959655-d50c53fc-4337-11e8-9806-b1bb9023b307.gif)
([demo in webm](https://webmshare.com/Gm48E))

The code I'm using:
```javascript
function foo(x) {
  return x == null;
}

function bar(x) {
  const shouldCrash = foo(x);
  if (shouldCrash)
    crash();
}

setTimeout(() => { bar(null); });
```

I'm running this script with:
```bash
nvs run chakracore-nightly --record index.js
```

My launch.json:
```javascript
{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Trace Debug",
      "type": "node",
      "request": "launch",
      "stopOnEntry": true,
      "program": "${workspaceRoot}/index.js",
      "runtimeExecutable": "${env:HOME}/.nvs/nvs",
      "runtimeArgs": [
        "run",
        "chakracore-nightly"
        "--nolazy",
        "--break-first",
        "--replay-debug=${workspaceRoot}/_diagnosticTraces/emitOnException_pid28716"
      ],
      "console": "internalConsole"
    }
  ]
}
```

I tried to set breakpoints inside `bar`, but it didn't worked. ~I also tried to use the `debugger` statement with no success~. I'm not sure if I'm doing something wrong or if this is a bug.

Edit: I was able to make it work with the `debugger` statement inside `foo()`. Now I'm not sure if this is by design (if it is, feel free to close this issue).

**Comment by kfarnung** [comment]
@mrkmarron Can you investigate?

**Comment by mrkmarron** [comment]
Hi @mmarchini, I took a quick look on my Mac (same config as you report). In the launch config you provide a "program" argument for the trace which is causing an issue in the setup of the trace debugger code. If you remove that line things should work as expected. 
```
{
      "name": "Trace Debug",
...
      "program": "${workspaceRoot}/index.js",
...
}
```
We are hijacking the Node launch config here and we know this can create some invalid configurations. I thought we had checking to warn for this particular case but I will investigate more. 

To make it easier to get going we just built a vscode extension [here](https://marketplace.visualstudio.com/items?itemName=ttd-trace-tools.node-chakracore-time-travel-debugger) that allows time-travel without needing to do an explicit recording first, downloads all the binaries for you, and that has a simple launch configuration provider.

Thanks and let us know if this takes care of the problem!
